bin/tflocal

```diff
@@ -247,6 +249,9 @@ def generate_s3_backend_config() -> str:
         get_or_create_bucket(configs["bucket"])
         get_or_create_ddb_table(configs["dynamodb_table"], region=configs["region"])
     result = TF_S3_BACKEND_CONFIG
+    if is_tf_legacy:
```

Comment:
I looked at that, and if I understand properly, you are suggesting to change the configuration of the aws provider from `s3_use_path_style` supported in [provider](https://registry.terraform.io/providers/hashicorp/aws/latest/docs#s3_use_path_style) `>4`  to `s3_force_path_style` for [providers version](https://registry.terraform.io/providers/hashicorp/aws/3.76.1/docs#s3_force_path_style) `<4`.

It can be quite the complex problem to define which version of a provider will be used as all modules are looked at by terraform before deciding on the providers version.

Let me know if I am missing something :thinking: